modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -395,90 +508,79 @@ cv::gimpl::ie::GIEExecutable::GIEExecutable(const ade::Graph &g,
-    // FIXME: Check input/output objects against GIsland protocol
+    GAPI_Assert(cv::util::holds_alternative<cv::GRunArgs>(in_msg));
+    const auto in_vector = cv::util::get<cv::GRunArgs>(in_msg);
 
-    for (auto& it : input_objs)   magazine::bindInArg (m_res, it.first, it.second);
-    for (auto& it : output_objs)  magazine::bindOutArg(m_res, it.first, it.second);
+    // (2) Collect inputs/outputs
+    input_objs.reserve(in_desc.size());
+    output_objs.reserve(out_desc.size());
+    for (auto &&it: ade::util::zip(ade::util::toRange(in_desc),
+                                   ade::util::toRange(in_vector)))
+    {
+        input_objs.emplace_back(std::get<0>(it), std::get<1>(it));
+    }
+    for (auto &&it: ade::util::indexed(ade::util::toRange(out_desc)))
+    {
+        output_objs.emplace_back(ade::util::value(it),
+                              out.get(ade::util::checked_cast<int>(ade::util::index(it))));
+    }
 
     // FIXME: Running just a single node now.
     // Not sure if need to support many of them, though
     // FIXME: Make this island-unmergeable?
     const auto &op = m_gm.metadata(this_nh).get<Op>();
 
-    // Initialize kernel's execution context:
-    // - Input parameters
-    IECallContext context;
-    context.args.reserve(op.args.size());
-    using namespace std::placeholders;
-    ade::util::transform(op.args,
-                          std::back_inserter(context.args),
-                          std::bind(&GIEExecutable::packArg, this, _1));
-
-    // NB: Need to store inputs shape to recognize GFrame/GMat
-    ade::util::transform(op.args,
-                         std::back_inserter(context.in_shapes),
-                         [](const cv::GArg& arg) {
-                             return arg.get<cv::gimpl::RcDesc>().shape;
-                         });
-    // - Output parameters.
-    for (const auto out_it : ade::util::indexed(op.outs)) {
-        // FIXME: Can the same GArg type resolution mechanism be reused here?
-        const auto out_port  = ade::util::index(out_it);
-        const auto out_desc  = ade::util::value(out_it);
-        context.results[out_port] = magazine::getObjPtr(m_res, out_desc);
-    }
+    // (3) Create kernel context
+    auto context = std::make_shared<IECallContext>(op.args,
+                                                   op.outs,
+                                                   std::move(input_objs),
+                                                   std::move(output_objs));
 
-    // And now trigger the execution
     GConstGIEModel giem(m_g);
     const auto &uu = giem.metadata(this_nh).get<IEUnit>();
     const auto &kk = giem.metadata(this_nh).get<IECallable>();
-    kk.run(this_iec, uu, context);
 
-    for (auto &it : output_objs) magazine::writeBack(m_res, it.first, it.second);
+    // (4) Only single async request is supported now,
+    // so need to wait until the kernel previos is over.
+    m_sync.wait();
```

Comment:
So what if there's no previous one? What if it is a first request of the execution sequence?

Reply:
Since acquire wasn't called, it won't hangs there and finish execution. (See SyncPrim implementation).
Does it make sense to rename this method ?

Reply:
Got it, thanks